### DISCUSSION TITLE
setData not updating the UI

### DISCUSSION THREAD

**Discussion by jackkemmish** [tinymce/tinymce#9782] [General]
Hi there, I am trying to do the below to add text to a field, I can see it's adding to `dialogApi.getData()` but it's not visually updating the UI:
```javascript
onChange: (dialogApi, details) => {
  dialogApi.setData({text: 'James'})
},
```
I can see according to the docs this is meant to update the data set, which it does if I log it to the console, however it's not updating the text field value on the UI.

Any ideas??

Thanks,
Jack

**Comment by dchubad** [comment]
Have you tried clearing your browser's cache then refreshing the page?

**Reply by jackkemmish** [reply]
Hey! Thanks for the reply.

I was tinkering a lot over night and have come to the realisation I am trying to do all this while using the `redial()` method as I need to change some fields based on user selection, and so it's clearing the value before it even sets lol.

I tried to use local storage to store my text and then pass it in but I don't think it's possible :/
